feat: Add info about plugin remapping and update paperweight version.

Leguan16 opened the pull request:
Add section about plugin remapping and system property to disable it.

Also updated paperweight version to 1.6.2

Leguan16 commented:
I seem to understand it now. I was using groovy and not kotlin dls. Thats why it didn't work. I got it working and it seems to be clear now whats going on :)

zlataovce commented on docs/paper/dev/getting-started/userdev.mdx:
````diff
@@ -103,3 +109,27 @@ tasks.assemble {
+
+:::
+
+```kotlin title="build.gadle.kts"
````
```suggestion
```kotlin title="build.gradle.kts"
```

zlataovce commented on docs/paper/dev/getting-started/userdev.mdx:
````diff
@@ -103,3 +109,27 @@ tasks.assemble {
+paperweight.reobfArtifactConfiguration = io.papermc.paperweight.userdev.ReobfArtifactConfiguration.MOJANG_PRODUCTION
+```
+
+additionally you need to remove all `dependsOn(reobfJar)` lines.
````
```suggestion
Additionally, you need to remove all `dependsOn(reobfJar)` lines.
```

zlataovce commented on docs/paper/dev/getting-started/userdev.mdx:
```diff
@@ -103,3 +109,27 @@ tasks.assemble {
+* By default, all Spigot/Bukkit plugins will be assumed to be Spigot-mapped if they do not specify their mappings namespace in the manifest. Conversely, all Paper plugins will be assumed to be Mojang-mapped if they do not specify their mappings namespace in the manifest.
+* Spigot-mapped plugins will need to be deobfuscated on first load, Mojang-mapped plugins will not.
+
+If you want to map your pluign to Mojang-mappings you need to add the following code to your build script:
```
```suggestion
If you want to map your plugin with Mojang mappings, you need to add the following code to your build script:
```

zlataovce commented on docs/paper/dev/getting-started/userdev.mdx:
```diff
@@ -103,3 +109,27 @@ tasks.assemble {
+
+Most of this process is done automatically by paperweight but there are some important things to know when using NMS from now on.
+
+### Default Mappings Assumption
```
```suggestion
### Default mappings assumption
```

zlataovce commented on docs/paper/dev/getting-started/userdev.mdx:
```diff
@@ -103,3 +109,27 @@ tasks.assemble {
 
+## 1.20.5
+As of 1.20.5 Paper will be shipping with a Mojang-mapped runtime instead of reobfuscating the server to Spigot mappings as we have traditionally done since Paper 1.17.
+Additionally. CraftBukkit classes will no longer be relocated into a versioned package.
```
```suggestion
Additionally, CraftBukkit classes will no longer be relocated into a versioned package.
```

zlataovce commented on docs/paper/dev/getting-started/userdev.mdx:
````diff
@@ -103,3 +109,27 @@ tasks.assemble {
 ```
 
+## 1.20.5
+As of 1.20.5 Paper will be shipping with a Mojang-mapped runtime instead of reobfuscating the server to Spigot mappings as we have traditionally done since Paper 1.17.
````
```suggestion
As of 1.20.5, Paper will be shipping with a Mojang-mapped runtime instead of reobfuscating the server to Spigot mappings as we have traditionally done since Paper 1.17.
```

also, hasn't the server been Spigot-mapped ever since Paper was created, not only since 1.17?

zlataovce commented on docs/paper/dev/getting-started/userdev.mdx:
```diff
@@ -24,6 +24,12 @@ This can make it hard to work with in a development environment. This plugin let
+:::caution[1.20.5]
+
+As of Minecraft version 1.20.5 Paper will be shipped with a Mojang-mapped runtime instead of reobfuscating the server to Spigot mappings as we have traditionally done since Paper 1.17.
+See [Mojang-Mappings](#1205) for more details.
```
```suggestion
See [here](#1205) for more details.
```

zlataovce commented on docs/paper/dev/getting-started/userdev.mdx:
```diff
@@ -24,6 +24,12 @@ This can make it hard to work with in a development environment. This plugin let
 
+:::caution[1.20.5]
+
+As of Minecraft version 1.20.5 Paper will be shipped with a Mojang-mapped runtime instead of reobfuscating the server to Spigot mappings as we have traditionally done since Paper 1.17.
```
```suggestion
As of Minecraft version 1.20.5, Paper will be shipped with a Mojang-mapped runtime instead of reobfuscating the server to Spigot mappings as we have traditionally done since Paper 1.17.
```

zlataovce commented on docs/paper/dev/getting-started/project-setup.mdx:
```diff
@@ -187,6 +187,49 @@ If everything went well, you should see something like this:
+    }
+}
+
+//if you have shadowJar configured
```
```suggestion
// if you have shadowJar configured
```

zlataovce commented on docs/paper/dev/getting-started/userdev.mdx:
```diff
@@ -103,3 +109,27 @@ tasks.assemble {
+* By default, all Spigot/Bukkit plugins will be assumed to be Spigot-mapped if they do not specify their mappings namespace in the manifest. Conversely, all Paper plugins will be assumed to be Mojang-mapped if they do not specify their mappings namespace in the manifest.
+* Spigot-mapped plugins will need to be deobfuscated on first load, Mojang-mapped plugins will not.
+
+If you want to map your plugin to Mojang-mappings you need to add the following code to your build script:
```
```suggestion
If you want to map your plugin with Mojang mappings, you need to add the following code to your build script:
```

zlataovce commented on docs/paper/dev/getting-started/project-setup.mdx:
````diff
@@ -187,6 +187,49 @@ If everything went well, you should see something like this:
+
+:::
+
+```groovy title="build.gradle(.kts)"
````
all build scripts in the docs are afaik Kotlin DSL, so specifying the Groovy variant is unnecessary
```suggestion
```kotlin title="build.gradle.kts"
```

zlataovce commented on docs/paper/dev/getting-started/project-setup.mdx:
```diff
@@ -187,6 +187,49 @@ If everything went well, you should see something like this:
+
+:::note[Paper plugins]
+
+if you are using paper plugins this step is not needed as plugins will be assumed to be Mojang-mapped.
```
```suggestion
If you are using Paper plugins, this step is not needed as plugins will be assumed to be Mojang-mapped.
```

zlataovce commented on docs/paper/dev/getting-started/project-setup.mdx:
```diff
@@ -187,6 +187,49 @@ If everything went well, you should see something like this:
+
+As of 1.20.5, Paper will be shipping with a Mojang-mapped runtime instead of reobfuscating the server to Spigot mappings.
+If you are using Spigot/Bukkit plugins, your plugin will automatically be remapped to Spigot mappings.
+This means that the server will have to deobfuscate and remap the plugin jar every time it is loaded for the first time.
```
```suggestion
This means that the server will have to deobfuscate and remap the plugin JAR every time it is loaded for the first time.
```

kennytv commented on docs/paper/dev/getting-started/project-setup.mdx:
```diff
@@ -187,6 +187,49 @@ If everything went well, you should see something like this:
+
+:::note[Paper plugins]
+
+If you are using Paper plugins, this step is not needed as plugins will be assumed to be Mojang-mapped.
```
You may still set the spigot namespace to get remapped

kennytv commented on docs/paper/dev/getting-started/project-setup.mdx:
```diff
@@ -187,6 +187,49 @@ If everything went well, you should see something like this:
 
 ![](assets/paper-plugin-overview.png)
 
+## Plugin remapping
```
This should also mention setting the attribute automatically via userdev

kennytv commented on docs/paper/dev/getting-started/project-setup.mdx:
```diff
@@ -187,6 +187,49 @@ If everything went well, you should see something like this:
 
+## Plugin remapping
+
+As of 1.20.5, Paper will be shipping with a Mojang-mapped runtime instead of reobfuscating the server to Spigot mappings.
```
ships with, same in the other two places

Leguan16 replied:
I just took jmp's message as reference. Will change it. Thank you for the feedback.

e-im commented on docs/paper/dev/getting-started/userdev.mdx:
```diff
@@ -24,6 +24,12 @@ This can make it hard to work with in a development environment. This plugin let
 
+:::caution[1.20.5]
+
+As of Minecraft version 1.20.5, Paper ships with a Mojang-mapped runtime instead of reobfuscating the server to Spigot mappings as we have traditionally done since Paper 1.17.
```
I'm not sure the since 1.17 part is relevant. userdev users don't need to know or care about Paper's internal development process. Might add more confusion without any real benefit.

Leguan16 commented on docs/paper/dev/getting-started/project-setup.mdx:
```diff
@@ -187,6 +187,49 @@ If everything went well, you should see something like this:
+## Plugin remapping
+
+As of 1.20.5, Paper will be shipping with a Mojang-mapped runtime instead of reobfuscating the server to Spigot mappings.
+If you are using Spigot/Bukkit plugins, your plugin will automatically be remapped to Spigot mappings.
```
oops 😅

e-im commented on docs/paper/dev/getting-started/project-setup.mdx:
```diff
@@ -187,6 +187,49 @@ If everything went well, you should see something like this:
+
+As of 1.20.5, Paper will be shipping with a Mojang-mapped runtime instead of reobfuscating the server to Spigot mappings.
+If you are using Spigot/Bukkit plugins, your plugin will automatically be remapped to Spigot mappings.
+This means that the server will have to deobfuscate and remap the plugin JAR every time it is loaded for the first time.
```
Maybe "on change" or something would be better.

Leguan16 replied:
on first load.

Leguan16 commented on docs/paper/admin/reference/system-properties.md:
```diff
@@ -210,3 +210,8 @@ It also remaps plugin CB calls to remove the version information.
 - **default**: `false`
 - **description**: Disables Paper's automatic upgrading of commands, including items with custom data defined in command blocks and other places that may contain commands, to the new component format introduced in version 1.20.5.
+
+#### paper.disable-plugin-rewriting
+
+- **default**: `false`
+- **description**: Disables plugin remapping and everything involving automatic plugin conversion by commodore.
```
I will think of something better. If someone has a suggestion please let me know.

olijeffers0n commented on docs/paper/dev/getting-started/userdev.mdx:
````diff
@@ -103,3 +109,37 @@ tasks.assemble {
 }
 ```
 
+## 1.20.5
````
Newlines after headings please. Same with subsequent Headings

Leguan16 commented:
I addressed the comments but i think I messed up with the rebase? 

For the system property we might want to add the forum post if it's going to mention it as well.

olijeffers0n commented on docs/paper/dev/getting-started/project-setup.mdx:
````diff
@@ -187,6 +187,93 @@ If everything went well, you should see something like this:
+
+:::
+
+```kotlin title="build.gradle.kts"
+tasks.jar {
+    manifest {
+        attributes["paperweight-mappings-namespace"] = "mojang"
+    }
+}
+
+// if you have shadowJar configured
+tasks.shadowJar {
+    manifest {
+        attributes["paperweight-mappings-namespace"] = "mojang"
+    }
+}
+```
+
+```xml title="pom.xml"
+<plugin>
+    <groupId>org.apache.maven.plugins</groupId>
+    <artifactId>maven-jar-plugin</artifactId>
+    <version>3.4.1</version>
+    <configuration>
+        <archive>
+            <manifestEntries>
+                <paperweight-mappings-namespace>mojang</paperweight-mappings-namespace>
+            </manifestEntries>
+        </archive>
+    </configuration>
+</plugin>
+```
````
Can you put these in a tab pane? See Velocity creating-your-first-plugin for info

Leguan16 replied:
👍🏻

Leguan16 replied:
done

olijeffers0n commented on docs/paper/dev/getting-started/project-setup.mdx:
````diff
@@ -187,6 +187,93 @@ If everything went well, you should see something like this:
+
+If you explicitly want to tell the server that your plugin is Spigot-mapped, you need to add the following code to your build script:
+
+```kotlin title="build.gradle.kts"
````
Ditto
